cli/cli-meta/src/index.ts

```diff
@@ -43,3 +43,26 @@ export function detectIfCurrentPkgIsExecutable (proc: NodeJS.Process = process):
 export function isExecutedByCorepack (env: NodeJS.ProcessEnv = process.env): boolean {
   return env.COREPACK_ROOT != null
 }
+
+export function getCurrentPackageName (): string {
+  return detectIfCurrentPkgIsExecutable() ? getExePackageName() : 'pnpm'
+}
+
+function getExePackageName (): string {
+  return `@pnpm/${normalizePlatformName()}-${normalizeArchName()}`
+}
+
+function normalizePlatformName (): string {
+  switch (process.platform) {
+  case 'win32': return 'win'
+  case 'darwin': return 'macos'
+  default: return process.platform
+  }
+}
+
+function normalizeArchName (): string {
+  if (process.platform === 'win32' && process.arch === 'ia32') {
+    return 'x86'
+  }
+  return process.arch
+}
```

Comment:
If all the functions here take parameter `process: { arch: NodeJS.Architecture, platform: NodeJS.Platform }`, `getCurrentPackageName` can be unit tested.